app/models/factory.rb

```diff
@@ -107,4 +76,30 @@ def family
   def to_s
     name + ":" + traits.map(&:name).join(",").to_s
   end
+
+  private
+    # Return same name and same traits having factory.
+    # TODO: improve algorithm
+    def self.search_same_factory(name, trait_names)
+      trait_names = trait_names.to_s
+      # This is maybe too slow
+      REDIS.smembers("factory_with_traits").find do |fwt|
+        factory_hash = JSON.parse(fwt)
+        return factory_hash if factory_hash["factory_name"] == name && factory_hash["traits"] == trait_names
+      end
+    end
+
+    # Return same factory. If if it is no asoociarion, overwrite new assos.
+    def self.overwrite_same_factory_asso(assos, same_factory)
+      if assos == same_factory.assos
+        return same_factory
+      elsif !assos.empty? && same_factory.assos.empty?
+        Asso.create_new_assos_and_relations(same_factory, assos)
+      end
+      same_factory.reload
+    end
+
+    def self.is_first_look_factory?
```

Comment:
Style/IndentationConsistency: Inconsistent indentation detected.<br>Style/PredicateName: Rename `is_first_look_factory?` to `first_look_factory?`.